### PR TITLE
Add sorting to gift card list

### PR DESCRIPTION
**Pull request by Cloud11PL**
I want to merge this change because I want to add sorting functionality to gift cards list. 

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://feature-gift-cards-post-mvp.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3905 Add sort to Gift Cards list](https://app.clickup.com/t/2549495/SALEOR-3905)

**Review by orzechdev — APPROVED** [review]

File: src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx [hunk 1]
```diff
@@ -65,23 +80,37 @@ export const GiftCardsListProvider: React.FC<GiftCardsListProviderProps> = ({
...
+    [params, settings.rowNumber]
   );
 
+  const handleGiftCardListError = (error: ApolloError) => {
+    const { message } = error.graphQLErrors[0];
+    notify({
+      status: "error",
+      text: message
+    });
+  };
```
Comment (dominik-zeglen) [inline src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx:100]: I don't think we should be handling errors in backend that originate from not out incorrect input but (probably) bug in application. What errors can we expect here?

Comment (Cloud11PL) [reply src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx:100]: Backend sends an error message that it is required to have a currently filter applird to sort by balance. For now there's no other place to add error handling like that.

File: src/giftCards/GiftCardsList/providers/GiftCardListDialogsProvider/hooks/useGiftCardListSort.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+type GiftCardSortListProps = Pick<
+  GiftCardsListConsumerProps,
+  "onSort" | "sort"
+>;
+
+const useGiftCardListSort = (): GiftCardSortListProps => {
+  const { onSort, sort } = useContext(GiftCardsListContext);
+
+  return { onSort, sort };
+};
```
Comment (dominik-zeglen) [inline src/giftCards/GiftCardsList/providers/GiftCardListDialogsProvider/hooks/useGiftCardListSort.ts:17]: Don't we already have wrapper around `useContext(GiftCardsListContext)`?

Comment (Cloud11PL) [reply src/giftCards/GiftCardsList/providers/GiftCardListDialogsProvider/hooks/useGiftCardListSort.ts:17]: Technically yes but I don't think exporting `onSort, sort` from these hooks (`useGiftCardList` and `useGiftCardListBulkActions`) is the best solution

Comment (dominik-zeglen) [reply src/giftCards/GiftCardsList/providers/GiftCardListDialogsProvider/hooks/useGiftCardListSort.ts:17]: I don't see why, can you elaborate?

Comment (Cloud11PL) [reply src/giftCards/GiftCardsList/providers/GiftCardListDialogsProvider/hooks/useGiftCardListSort.ts]: @dominik-zeglen I just didn't think that adding `sort, onSort` to the main gift card hook would look clean (one hook do to it all) but I think you're right and it's fine.

File: src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx (hunk 1 above)
Comment (Cloud11PL) [reply src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx]: @dominik-zeglen Do you recommend a different approach to this problem?

Comment (dominik-zeglen) [reply src/giftCards/GiftCardsList/providers/GiftCardListProvider/GiftCardListProvider.tsx]: Not really, lets leave it here

**Review by dominik-zeglen — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**